src/Test/L1/Plugins/FakeCheckoutTask.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using Agent.Plugins.Repository;
+using Agent.Sdk;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.L1.Worker
+{
+    public class FakeCheckoutTask : CheckoutTask
+    {
+        public override async Task RunAsync(AgentTaskPluginExecutionContext executionContext, CancellationToken token)
+        {
+            SourceProviderFactory = new FakeSourceProviderFactory();
+            await base.RunAsync(executionContext, token);
+        }
+    }
+
+    public sealed class FakeSourceProviderFactory : SourceProviderFactory
+    {
+        public override ISourceProvider GetSourceProvider(string repositoryType)
+        {
+            ISourceProvider sourceProvider = base.GetSourceProvider(repositoryType);
+            if (sourceProvider.GetType() == typeof(GitHubSourceProvider))
+            {
+                return new FakeGitHubSourceProvider();
+            }
+            else if (sourceProvider.GetType() == typeof(BitbucketGitSourceProvider))
+            {
+                return new FakeBitbucketGitSourceProvider();
+            }
+            else if (sourceProvider.GetType() == typeof(ExternalGitSourceProvider))
+            {
+                return new FakeExternalGitSourceProvider();
+            }
+            else if (sourceProvider.GetType() == typeof(TfsGitSourceProvider))
+            {
+                return new FakeTfsGitSourceProvider();
+            }
+            else
+            {
+                throw new Exception("Source provider not mocked: " + repositoryType);
```

Comment:
We initially chose to just mock Git source providers, should be easily extensible if we decide we want to write Tfvc or svn l1s